File: Tutorial3-2Scope2/src/main/java/com/example/tutorial3scope2/MyApplication.java
package com.example.tutorial3scope2;

import android.app.Application;

import com.example.tutorial3scope2.di.DaggerMyComponent;
import com.example.tutorial3scope2.di.MyComponent;
import com.example.tutorial3scope2.di.SharedPreferencesModule;


public class MyApplication extends Application {
    private MyComponent myComponent;

    @Override
    public void onCreate() {
        super.onCreate();

        myComponent = DaggerMyComponent.builder()
                .sharedPreferencesModule(new SharedPreferencesModule(getApplicationContext()))
                .build();

    }

    public MyComponent getMyComponent() {
        return myComponent;
    }
}

File: Tutorial3-2Scope2/src/main/java/com/example/tutorial3scope2/MyApplication.kt
package com.example.tutorial3scope2

import android.app.Application
import com.example.tutorial3scope2.di.ApplicationComponent
import com.example.tutorial3scope2.di.ApplicationModule
import com.example.tutorial3scope2.di.DaggerApplicationComponent

class MyApplication : Application() {

    lateinit var applicationComponent: ApplicationComponent
        private set

    override fun onCreate() {
        super.onCreate()
        applicationComponent = DaggerApplicationComponent.builder()
                .applicationModule(ApplicationModule(applicationContext))
                .build()
    }

}